controllers/infrastructure/byomachine_controller_test.go

```diff
@@ -117,6 +118,32 @@ var _ = Describe("Controllers/ByomachineController", func() {
 			Expect(err).To(MatchError("nodes \"" + byoHost.Name + "\" not found"))
 		})
 
+		Context("When node.Spec.ProviderID is already set", func() {
+
+			BeforeEach(func() {
+				byoHost = builder.ByoHost(defaultNamespace, "test-node-providerid-host").Build()
+				Expect(k8sClientUncached.Create(ctx, byoHost)).Should(Succeed())
+			})
+
+			It("should not return error when node.Spec.ProviderID is with correct value", func() {
+				node := builder.Node(defaultNamespace, byoHost.Name).Build()
+				node.Spec.ProviderID = fmt.Sprintf("%s%s/%s", controllers.ProviderIDPrefix, byoHost.Name, util.RandomString(controllers.ProviderIDSuffixLength))
```

Comment:
I think it will be useful to add a method in the builder package to set provider id. Code will be more cleaner.

Reply:
ok.